packages/tupaia-web/src/features/Map/utils/useDefaultMapOverlay.ts

```diff
@@ -32,14 +34,27 @@ export const useDefaultMapOverlay = (
     const isValidMapOverlayId = !!mapOverlaysByCode[selectedMapOverlay!];
 
     if (!selectedMapOverlay || !isValidMapOverlayId) {
-      const defaultMapOverlayId = project.defaultMeasure || DEFAULT_MAP_OVERLAY_ID;
+      const { defaultMeasure } = project;
+      const overlayCodes = mapOverlaysByCode ? Object.keys(mapOverlaysByCode) : [];
+      // If there are no map overlays, the default overlay is the DEFAULT_MAP_OVERLAY_ID
+      // however if there are map overlays, use the first available overlay
+      let defaultMapOverlayId = overlayCodes.length > 0 ? overlayCodes[0] : DEFAULT_MAP_OVERLAY_ID;
```

Comment:
Thanks for picking this up - looks good!

Reply:
Oh actually I think it's still not quite right. A default map overlay is getting assigned when there is no map overlay data and I don't think that ever makes sense. What about this?

```
useEffect(() => {
    const isValidMapOverlayId = !!mapOverlaysByCode[selectedMapOverlay!];
    const overlayCodes = mapOverlaysByCode ? Object.keys(mapOverlaysByCode) : [];

    if (!project || overlayCodes.length === 0) {
      return;
    }

    if (!selectedMapOverlay || !isValidMapOverlayId) {
      const { defaultMeasure } = project;

      if (mapOverlaysByCode[defaultMeasure]) {
        urlSearchParams.set(URL_SEARCH_PARAMS.MAP_OVERLAY, defaultMeasure);
      }

      if (mapOverlaysByCode[DEFAULT_MAP_OVERLAY_ID]) {
        urlSearchParams.set(URL_SEARCH_PARAMS.MAP_OVERLAY, DEFAULT_MAP_OVERLAY_ID);
      }
    }

    if (!selectedMapOverlayPeriod) {
      urlSearchParams.set(URL_SEARCH_PARAMS.MAP_OVERLAY_PERIOD, DEFAULT_PERIOD_PARAM_STRING);
    }

    // we have to navigate using navigate instead of setUrlParams because setUrlParams seems to override any existing params and existing hash
    navigate({
      ...location,
      search: urlSearchParams.toString(),
    });
  }, [JSON.stringify(mapOverlaysByCode), project, selectedMapOverlay]);